lib/yt/request.rb

```diff
@@ -200,14 +200,18 @@ def run_again?
 
     # Returns the list of server errors worth retrying the request once.
     def server_errors
-      [
+      errors = [
         OpenSSL::SSL::SSLError,
-        OpenSSL::SSL::SSLErrorWaitReadable,
         Errno::ETIMEDOUT,
         Errno::ENETUNREACH,
         Errno::ECONNRESET,
         Net::HTTPServerError
       ]
+      # OpenSSL::SSL::SSLErrorWaitReadable does not exist in Ruby 1.9.3
+      if defined?(OpenSSL::SSL::SSLErrorWaitReadable)
+        errors << OpenSSL::SSL::SSLErrorWaitReadable
+      end
+      errors
     end
```

Comment:
@bodrovis I'm sorry for not accepting this PR before, I think this is the correct way.

However, do you mind changing the condition to `if RUBY_VERSION > '2'`? That's really the implicit requirement that we are making and that we are confirming in the tests. So I would like the method above to be:

```ruby
def server_errors
  [
    OpenSSL::SSL::SSLError,
    Errno::ETIMEDOUT,
    Errno::ENETUNREACH,
    Errno::ECONNRESET,
    Net::HTTPServerError
  ] + (RUBY_VERSION < '2' ? [] : [OpenSSL::SSL::SSLErrorWaitReadable])
end
```

If you agree and change your commit to that, I will merge right away :dancers: